src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -121,6 +145,36 @@ void GcsServer::Start() {
   gcs_init_data->AsyncLoad([this, gcs_init_data] { DoStart(*gcs_init_data); });
 }
 
+void GcsServer::CacheAndSetServerToken() {
+  static std::string const kTokenNamespace = "cluster";
+  static std::string const kTokenKey = "server_uuid";
+  switch (storage_type_) {
+    case StorageType::IN_MEMORY:
+      token_promise_.set_value(GenerateUUIDV4());
```

Comment:
how about we introduce class concept cluster_id into Ray?
https://sourcegraph.com/github.com/ray-project/ray/-/blob/src/ray/common/id.h